share/misc/Makefile

```diff
@@ -18,4 +18,15 @@ FILES+=	usb_hid_usages
 FILES+=	usbdevs
 .endif
 
+FILES+=	perf_warnings
+.PHONY: perf_warnings
+perf_warnings:
+	echo "# motd performance warnings, generated by share/misc/Makefile" > ${.TARGET}
+.ifdef MALLOC_DEBUG
+	echo "WARNING: build defined MALLOC_DEBUG, expect reduced performace" >> ${.TARGET}
```

Comment:
I'd just have this file contain a list of options and then append_perf_warnings can template it just like it does for kernel warnings? Avoids repetition here.

Also is there a reason why you went with "WARNING: build defined X" rather than "WARNING: X option enabled" for world? They're still options, and called such in places. Would be simplifying code-wise (just add `grep -v '^#' perf_warnings` to _warn_options) and look more uniform if they all used the same message.

Reply:
I was trying to make a distinction between kernel and user space options. Not sure how much sense it makes.

Reply:
I can see the value in that, but I don't think this communicates that at all to someone who doesn't already know what all these things are, an option could easily be for world and a build define for kernel. If you want a distinction I'd suggest something uniform like "WARNING: FOO (kernel|world) option defined, expect reduced performance".